picard/ui/find.py

```diff
@@ -0,0 +1,164 @@
+
+        checkboxes = {}
+        for file_filter in ["filename", "filepath"]:
+            checkbox = QtWidgets.QCheckBox(_(self.file_filters[file_filter]), scroll_content)
+            checkbox.setChecked(file_filter in self.selected_filters)
+            scroll_layout.addWidget(checkbox)
+            checkboxes[file_filter] = checkbox
+
+        scroll_layout.addItem(QtWidgets.QSpacerItem(0, 10, QtWidgets.QSizePolicy.Policy.Minimum, QtWidgets.QSizePolicy.Policy.Fixed))
+
+        label = QtWidgets.QLabel(_("Tag Filters"), scroll_content)
+        scroll_layout.addWidget(label)
+
+        # Add a horizontal separator
+        line = QtWidgets.QFrame(scroll_content)
+        line.setFrameShape(QtWidgets.QFrame.Shape.HLine)
+        line.setFrameShadow(QtWidgets.QFrame.Shadow.Sunken)
+        scroll_layout.addWidget(line)
+
+        # Add checkboxes for all tags
+        for tag in sorted(self.valid_tags, key=lambda t: _(ALL_TAGS.display_name(t._name)).lower()):
+            checkbox = QtWidgets.QCheckBox(ALL_TAGS.display_name(tag._name), scroll_content)
+            checkbox.setChecked(tag._name in self.selected_filters)
+            scroll_layout.addWidget(checkbox)
+            checkboxes[tag._name] = checkbox
+
+        scroll_content.setLayout(scroll_layout)
+        scroll.setWidget(scroll_content)
+        layout.addWidget(scroll)
+
+        # Buttons
+        button_box = QtWidgets.QDialogButtonBox(
+            QtWidgets.QDialogButtonBox.StandardButton.Ok | QtWidgets.QDialogButtonBox.StandardButton.Cancel)
+        button_box.accepted.connect(dialog.accept)
+        button_box.rejected.connect(dialog.reject)
+        layout.addWidget(button_box)
+
+        # Show dialog and process result
+        if dialog.exec() == QtWidgets.QDialog.DialogCode.Accepted:
+            self.selected_filters = []
+            for tag, checkbox in checkboxes.items():
+                if checkbox.isChecked():
+                    self.selected_filters.append(tag)
+
+            if not self.selected_filters:
+                self.selected_filters = []
+
+            # Update button text
+            self.filter_button.setText(self.make_button_text(self.selected_filters))
+
+            self._query_changed(self.find_query_box.text())
+
+    @classmethod
+    def make_button_text(cls, selected_filters):
+        if not selected_filters:
+            return _("Filters")
+
+        if len(selected_filters) == 1:
+            if selected_filters[0] not in cls.file_filters.keys():
+                return _(ALL_TAGS.display_name(selected_filters[0]))
+            else:
+                return _(cls.file_filters[selected_filters[0]])
+
+        return _("{num} filters").format(num=len(selected_filters))
+
+    def _query_changed(self, text):
+        self.findChanged.emit(text, self.selected_filters)
+
+    def clear(self):
+        self.find_query_box.clear()
+        self.selected_filters = []
+        self.filter_button.setText(_("Filters"))
```

Comment:
Maybe store default button label in a property:

At class level, something like:

```python
    default_filter_button_label = N_("Filters")

    def set_filter_button_label(self, label=None):
        if label is None:
            label = _(default_filter_button_label)
        self.filter_button.setText(label)

    def clear(self):
        self.find_query_box.clear()
        self.selected_filters = []
        self.set_filter_button_label()
```

`make_button_text()` can then return `None` instead of `_("Filters")`